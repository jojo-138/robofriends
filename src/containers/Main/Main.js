import { useEffect } from 'react';
import 'tachyons';
import SearchBar from '../../components/SearchBar/SearchBar';
import CardList from '../../components/CardList/CardList';


const Main = ({ onFetchRobots, onSearchChange, robots, searchField }) => {
  useEffect (() => {
    onFetchRobots();
  }, [onFetchRobots])

  return (
    <>
      <div className='title f1 black-70'>Robofriends</div>
      <SearchBar handleChange={onSearchChange}/>
      <CardList users={robots} input={searchField}/>
    </>
  )
}

export default Main;