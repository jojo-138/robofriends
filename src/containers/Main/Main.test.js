import Main from "./Main";
import { render, screen } from "@testing-library/react";
import ShallowRenderer from 'react-test-renderer/shallow';
import userEvent from "@testing-library/user-event";

describe('Main container', () => {
  test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    expect(renderer.render(<Main />)).toMatchSnapshot();
  })

  test('onFetchRobots prop is called upon render and onSearchChange prop on when user types', () => {
    const mockOnFetchRobots = jest.fn();
    const mockOnSearchChange = jest.fn();
    const mockRobots = [
      {
        id: 1,
        name: 'john',
        email: 'john@gmail.com'
      },
      {
        id: 2,
        name: 'joana',
        email: 'joana@gmail.com'
      }
    ]

    render(<Main onFetchRobots={mockOnFetchRobots} onSearchChange={mockOnSearchChange} robots={mockRobots} searchField='' />)
    expect(mockOnFetchRobots).toBeCalledTimes(1);
    expect(mockOnSearchChange).not.toBeCalled();

    const input = screen.getByRole('textbox');

    userEvent.type(input, 'john');

    expect(mockOnSearchChange).toBeCalledTimes(4);
  })
})