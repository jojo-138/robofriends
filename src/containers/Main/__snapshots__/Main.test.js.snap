// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main container renders correctly 1`] = `
<React.Fragment>
  <div
    className="title f1 black-70"
  >
    Robofriends
  </div>
  <SearchBar />
  <CardList />
</React.Fragment>
`;
